ksqldb-parser/src/main/java/io/confluent/ksql/util/QueryMetaDataCollector.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import org.joda.time.DateTime;
+
+public final class QueryMetaDataCollector {
+  private static final Hashtable<QueryIdKey, String> queryIdTable = new Hashtable<>();
+  private static final Hashtable<String, String> structurallySimilarTable = new Hashtable<>();
+
+  private QueryMetaDataCollector() {}
+
+  public static Map<String, String> getMetaData(
+      final KsqlConfig config,
+      final String nonAnonQuery,
+      final String anonQuery)
+      throws Exception {
+
+    // get metadata from control plane
+    final Map<String, String> metaData = new HashMap<>();
+    final String clusterNamespace = config
+        .getString(KsqlConfig.KSQL_CCLOUD_QUERYANONYMIZER_CLUSTER_NAMESPACE);
+
+    // check if metadata contains org id and cluster id
+    if (clusterNamespace.length() == 0) {
+      throw new Exception("Organization Id or Cluster Id could not be found");
+    }
+
+    metaData.put("cluster_namespace", clusterNamespace);
+    metaData.put("id", getQueryId(nonAnonQuery, clusterNamespace));
+    metaData.put("structurally_similar_id", getStructurallySimilarId(anonQuery));
+    metaData.put("time_of_creation", DateTime.now().toString());
+
+    return metaData;
+  }
+
+  private static String getStructurallySimilarId(final String query) {
+    if (!structurallySimilarTable.containsKey(query)) {
```

Comment:
This could be just ```
return UUID.uuidV5(clusterNamespace, query)
```

https://en.wikipedia.org/wiki/Universally_unique_identifier#Versions_3_and_5_(namespace_name-based)

This might require you to actually turn the UUID to one of the compatible formats, but that just means changing it in a few places